src/classes/BDI_DataImportDeleteBTN_CTRL.cls

```diff
@@ -143,10 +152,20 @@ public with sharing class BDI_DataImportDeleteBTN_CTRL {
      * @return PageReference 
      */
     public PageReference close() {
-        String retUrl = ApexPages.currentPage().getParameters().get('retURL');
+        String uiTheme = UserInfo.getUiThemeDisplayed();
```

Comment:
If I read this right, you are basically ignoring the retURL parameter and instead assembling the return contextually? I wonder if there's a way to still use the return URL but swap out the base URL/domain to avoid redirects outside of the Salesforce instance. 

But - are we worried about hijacking within the instance? Technically I suppose someone could host a hostile VF page and redirect there. I'm just concerned that this might have gaps, especially if (god forbid) there's a new design system anytime soon.

Reply:
That's correct. One of the recommended ways of avoiding an open redirect like what the security team found is to explicitly specify where the redirect will take the user. So instead of using a more open `retURL`, not knowing what value that URL could contain, we would explicitly define it. This close function is only used in one place on the page and only ever redirects back to one page. I think what we're mainly concerned with regarding open redirect is sending the user outside of the internal instance to another URL entirely. 

Here is a [trailhead](https://trailhead.salesforce.com/en/content/learn/modules/secdev_application_logic_vulnerabilities/secdev_app_logic_preventing_open_redirect) that was linked on the WI that refers to the `hard-coded redirect` approach. I don't think doing the extra sanitation on `retURL` is necessary unless we were going to be dynamically redirecting users to different pages based on different actions.

Reply:
This makes sense - thanks for looping me in to that ongoing thread!